packages/docusaurus-mdx-loader/src/remark/rightToc/search.js

```diff
@@ -5,11 +5,29 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+// @ts-check
+
 const toString = require('mdast-util-to-string');
 const visit = require('unist-util-visit');
 const escapeHtml = require('escape-html');
 
+/** @typedef {import('@docusaurus/types').MarkdownRightTableOfContents} TOC */
+/** @typedef {import('unist').Node} Node */
+
+/**
+ * @typedef {Object} StringValuedNode
+ * @property {string} type
+ * @property {string} value
+ * @property {number} depth
+ * @property {Object} data
+ * @property {StringValuedNode[]} children
+ */
```

Comment:
Unfortunately, the `Node` type imported from `unist` doesn't seem to directly usable. The existing code seems to assume that the `value` field is always string. Therefore, I have to duplicate some type definition here.